src/metabase/driver/common/parameters/operators.clj

```diff
@@ -0,0 +1,79 @@
+                      :number/>=               :>=
+                      :number/<=               :<=})
+
+(def ^:private binary {:number/between :between})
+
+(def ^:private variadic {:string/=  :=
+                         :string/!= :!=
+                         :number/=  :=
+                         :number/!= :!=})
+
+(def ^:private all-ops (into #{} (mapcat keys [unary binary variadic])))
+
+(s/defn operator? :- s/Bool
+  "Returns whether param-type is an \"operator\" type."
+  [param-type]
+  (contains? all-ops param-type))
+
+(s/defn ^:private verify-type-and-arity
+  [field param-type param-value]
+  (letfn [(maybe-arity-error [n]
+            (when (not= n (count param-value))
+              (throw (ex-info (format "Operations Invalid arity: expected %s but received %s"
+                                      n (count param-value))
+                              {:param-type  param-type
+                               :param-value param-value
+                               :field-id    (second field)
+                               :type        qp.error-type/invalid-parameter}))))]
+    (cond (contains? unary param-type)    (maybe-arity-error 1)
+          (contains? binary param-type)   (maybe-arity-error 2)
+          (contains? variadic param-type) (when-not (seq param-value)
+                                            (throw (ex-info (format "No values provided for operator: %s" param-type)
+                                                            {:param-type  param-type
+                                                             :param-value param-value
+                                                             :field-id    (second field)
+                                                             :type        qp.error-type/invalid-parameter})))
+          :else                           (throw (ex-info (format "Unrecognized operation: %s" param-type)
+                                                          {:param-type  param-type
+                                                           :param-value param-value
+                                                           :field-id    (second field)
+                                                           :type        qp.error-type/invalid-parameter})))))
+
+(s/defn to-clause :- mbql.s/Filter
+  "Convert an operator style parameter into an mbql clause. Will also do arity checks and throws an ex-info with
+  `:type qp.error-type/invalid-parameter` if arity is incorrect."
+  [{param-type :type [a b :as param-value] :value [_ field :as _target] :target :as param}]
+  (verify-type-and-arity field param-type param-value)
+  (let [field' (params/wrap-field-id-if-needed field)]
+    (cond (contains? binary param-type)
+          [(binary param-type) field' a b]
+
+          (contains? unary param-type)
+          [(unary param-type) field' a]
+
+          (contains? variadic param-type)
+          (into [(variadic param-type) field'] param-value)
```

Comment:
since we verify type and arity tthis whole thing could essentially be this one line `(into [(translate param-type) field'] param-value)`. I left it as such in case we do need to do more transforms. Originally I was going to have a var of `negated` values and return `[:not (to-clause (update param :type negate))]` but mbql already had sugar for that (`:does-not-contain` -> `[:not [:contains ...]]`).